docs/docs/async-dialogs.md

```diff
@@ -4,3 +4,628 @@ title: Async Dialogs
 ---
 
 [async-dialogs](unfinished-article)
+
+Typically *promises* are used in conjunction with asynchronous tasks such as a
+network request or a `setTimeout`; a lesser explored use is in dealing with user
+input. Since a program has to wait for a user to continue some actions it makes
+sense to consider it an asynchronous event.
+
+For comparison I'll start with an example of a *synchronous* user interaction
+using the `window.prompt` and then move to an *asynchronous* interaction my
+making our own DOM based prompt. To begin here is a template for a simple HTML
+page:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Async Dislogs Example</title>
+  <script src="//cdn.jsdelivr.net/bluebird/3.0.5/bluebird.js"></script>
+  <script type="text/javascript">
+    document.addEventListener('DOMContentLoaded', function() {
+      var time = document.getElementById('time-stamp');
+      clockTick();
+      setInterval(clockTick, 1000);
+      function clockTick() {
+        time.innerHTML = '' + new Date();
+      }
+    });
+  </script>
+</head>
+<body>
+  <p>The current time is <span id="time-stamp"></span>.</p>
+  <p>Your name is <span id="prompt"></span>.</p>
+  <button id="action">Set Name</button>
+</body>
+</html>
+```
+
+`window.prompt` blocks the web page from processing while it waits for the user
+to enter in data. I has to block because the input is returned and the next line
+of code needs that result. But for sake of this tutorial we are going to convert
+the typical conditional code into a promise API.
+
+```javascript
+function promptPromise(message) {
+  return new Promise(function(resolve, reject) {
+    var result = window.prompt(message);
+    if (result != null) {
+      resolve(result);
+    } else {
+      reject();
+    }
+  });
+}
+
+document.addEventListener('DOMContentLoaded', function() {
```

Comment:
In general - it's considered a bad practice to add DOM ready events if your code is not being blindly included - you can safely include it at the end of the body section.